### ISSUE TITLE
add the pip installation strategy to the documentation

### ISSUE THREAD

**Issue by 12rambau** [gboeing/osmnx#820]
**Is your feature proposal related to a problem?**

I wanted to use the lib and was discovering the installation section of the documentation. The only documented way is `conda`, even though the lib is published and up-to-date on pipy. It didn't took me long to figure it out but that was still a missing information. 

**Describe the solution you'd like to propose**

I would like to update the documentation file to add the second way of installing osmnx: `pip`

**Comment by gboeing** [comment]
In a sense, we *do* document the pip installation strategy. The [docs](https://osmnx.readthedocs.io/en/stable/#installation) state:

> you can install OSMnx via pip if you already have all of its dependencies installed and fully tested on your system. Note: installing the dependencies with pip is nontrivial. If you don’t know exactly what you’re doing, just use conda as described above.

We don't document it any further than that, because there's nothing else to document on our end. OSMnx is pure-Python and trivially easy to install *if you have the dependencies all installed and tested*. The devil's in the details, and some of the dependencies' dependencies have C extensions and rely on other compiled libraries and require system-specific compilation, installation, and configuration. To install the dependencies, you have to go to each dependency's web site and follow their instructions, which is a nontrivial (and often-changing) process. If someone needs/wants to install OSMnx via pip, it is possible but difficult.

Long story short: we don't (and really can't) support pip installation, and don't document it accordingly.

**Comment by 12rambau** [comment]
thanks for taking the time to answer 👍

**Comment by 12rambau** [comment]
now that GDAL has been removed from the deps of osmnx (https://github.com/gboeing/osmnx/issues/916) does it means that a pure pip installation is possible again or is there still other C-like dependencies ?

**Comment by gboeing** [comment]
The [latest docs](https://osmnx.readthedocs.io/en/latest/installation.html#pip) note that pip is an installation option, but not "officially supported" by the OSMnx project because we can't personally guarantee the availability, compilation, and installation of C dependencies on every system. In other words, pip installation usually works on most systems, but if it it doesn't work on your system, you'll have to seek support from the relevant dependency that's not working.

It's worth noting that OSMnx has an optional dependency of rasterio, which itself depends on GDAL. So a full installation of OSMnx still installs GDAL, though it is no longer a direct requirement. Additionally, OSMnx requires shapely (which depends on GEOS) and GeoPandas (which depends on pyproj which in turn depends on PROJ). So there are still compiled sub-dependencies. Note that GeoPandas [uses similar](https://geopandas.org/en/stable/getting_started/install.html#installing-with-pip) language as us, noting that pip installation is possible but potentially nontrivial.
